linc_convert/utils/io/zarr/__init__.py

```diff
@@ -0,0 +1,27 @@
+"""ZarrIO module for handling Zarr data structures."""
+
+import warnings
+
+from .abc import ZarrArray, ZarrGroup, ZarrNode
+from .drivers.zarr_python import ZarrPythonArray, ZarrPythonGroup
+from .factory import from_config, open, open_group
+
+try:
+    import tensorstore as TS  # noqa: F401
+
+    from .drivers.tensorstore import ZarrTSArray, ZarrTSGroup
+except ImportError:
+    warnings.warn("Tensorstore is not installed, driver disabled")
+
+__all__ = [
+    ZarrArray,
+    ZarrGroup,
+    ZarrNode,
+    ZarrPythonArray,
+    ZarrPythonGroup,
+    from_config,
+    open,
+    open_group,
+    "ZarrTSArray",
+    "ZarrTSGroup",
+    ]
```

Comment:
When there are optional imports, I usually only add them to `__all__` if they are succesfully importer (not sure what best practices are). Also, I always only have strings in `__all__`, not references to actual objects.

So I would have 
```python
__all__ = ["Always1", "Always2"]
from .always import Always1, Always2
try:
    import OPTIONAL as _
    from .optionals import Optional1, Optional2
    __all__ += ["Optional1", "Optional2"]
except ImportError:
    warnings.warn("warn")
```